### DISCUSSION TITLE
Error 500 al exportar listas en Excel (.xlsx)

### DISCUSSION THREAD

**Discussion by rozepeda** [chamilo/chamilo-lms#5877] [General]
**Describe el problema**
Al intentar exportar listas en excel, el servidor entra en "error 500". (la exportación en formatos csv y pdf funcionan correctamente)

[19-Jul-2024 14:04:40 America/Mexico_City] PHP Fatal error:  Uncaught PHPExcel_Writer_Exception: Could not open C:/wamp/www/campus/app/cache/669ab8c8bf634.xlsx for writing. in C:\wamp\www\campus\vendor\phpoffice\phpexcel\Classes\PHPExcel\Writer\Excel2007.php:234
Stack trace:
#0 C:\wamp\www\campus\vendor\ddeboer\data-import\src\Writer\ExcelWriter.php(115): PHPExcel_Writer_Excel2007->save('C:/wamp/www/cam...')
#1 C:\wamp\www\campus\main\inc\lib\export.lib.inc.php(87): Ddeboer\DataImport\Writer\ExcelWriter->finish()
#2 C:\wamp\www\campus\main\user\user.php(407): Export::arrayToXls(Array, 'Lista de estudi...')
#3 {main}
  thrown in C:\wamp\www\campus\vendor\phpoffice\phpexcel\Classes\PHPExcel\Writer\Excel2007.php on line 234

**Cómo reproducir**
Pasos para reproducir el problema:
1. Ir a 'curso'
2. Dar clic en 'usuarios'
3. Dar clic en 'Exportar a un fichero XLS....'

**Comportamiento esperado**
Obtener un archivo excel con la lista de usuarios

**Capturas de pantalla**
Si aplicable, agrega capturas de pantalla para ayudar a ubicar el problema.

**Escritorio (por favor completa la información siguiente lo mejor posible):**
 - Sistema operativo : Windows 10 /Windows 11
 - Navegador: Chrome, Edge, Firefox

**Servidor
 - Sistema operativo : Windows Server 2019
 - Wampserver install version 3.3.6
 - Apache 2.4.41 Port 80 - PHP 7.4.33
 - MySQL 8.0.18 Port 3306
 - PHP 7.4.0 for CLI

**Contexto adicional**
Exportar a csv, pdf, funciona correctamente

**Comment by Kaneda-1** [comment]
Hola @rozepeda en el mensaje de error te indica
Could not open C:/wamp/www/campus/app/cache/669ab8c8bf634.xlsx for writing...

¿En el servidor Windows Server 2019 has revisado permisos de lectura/escritura en esa ruta Could not open C:/wamp/www/campus/app/cache para el usuario de Apache?

Un saludo.

**Comment by rozepeda** [comment]
Si, es posible escribir en ese directorio, de hecho, como comento, los csv se exportan sin problemas y esos tambien se escriben ahi, todos los temporales se pueden escribir y borrar mediante chamilo.

Sin embargo los xsl no se pueden generar y automaticamente manda el error 500..

😵
